network/kademlia.go

```diff
@@ -902,6 +903,68 @@ func (k *Kademlia) string() string {
 	return "\n" + strings.Join(rows, "\n")
 }
 
+func (k *Kademlia) MarshalJSON() (data []byte, err error) {
+	var (
+		kademliaInfo   = make(map[string]string)
+		depth          = depthForPot(k.conns, k.NeighbourhoodSize, k.base)
+		numConns       = k.conns.Size()
+		numAddrs       = k.addrs.Size()
+		connectedAddrs = make(map[string]struct{})
+		connBinMap     = make(map[string]string)
+		addrsBinMap    = make(map[string]string)
+	)
+
+	jsonizeAddrs := func(m map[string]string) func(bin *pot.Bin) bool {
+		return func(bin *pot.Bin) bool {
+			binMap := make(map[string]string)
+
+			po := bin.ProximityOrder
+			if po >= k.MaxProxDisplay {
+				po = k.MaxProxDisplay - 1 // why???
```

Comment:
I would also like to know why.

Reply:
it was in the original code (i.e. in the `string()` function) but i don't understand why